### ISSUE TITLE
You are wrongly logged out from "Firefox Notes" after reopening the sidebar if a sync action was previously performed

### ISSUE THREAD

**Issue by Softvision-MariusComan** [mozilla/notes#802]
**[Affected versions]:**
- Firefox 59.0.1 and above

**[Affected Platforms]:**
- All Windows
- All Mac
- All Linux

**[Prerequisites]:**
- Have a Firefox profile with the latest version of the "Firefox Notes" add-on (3.2.0dev custom built on 2018-03-26) installed.
- You are logged into "Firefox Notes".
- Have ~20 notes in the sidebar.

**[Steps to reproduce]:**
1. Open the browser with the profile from prerequisites.
2. Click the "Sync Your Notes" button.
3. Close and reopen the sidebar.
4. Observe the bottom toolbar.

**[Expected result]:**
- The "Synced at (hour) message is displayed.

**[Actual result]:**
- You are logged out and the "Sign in to Sync" message is displayed.

**[Notes]:**
- The issue is also reproducible if the sidebar is switched/closed while the "Syncing changes..." message is displayed.
- The issue is still reproducible after restart.
- Attached a screen recording of the issue: 
![rec of the issue](https://user-images.githubusercontent.com/31243259/37916807-793c439e-3125-11e8-9ad0-d882d678d798.gif)

**Comment by vladikoff** [comment]
Right, if there is an error with sync we don't show an error and then just lose the state, Ref: https://github.com/mozilla/notes/issues/791#issuecomment-375711775

**Comment by sebastienbarbier** [comment]
@ryanfeeley can you **suggest a UI** for an error notification ?  

We asked ourself where an alert would shows up :
![alert](https://user-images.githubusercontent.com/3018618/37941220-55dfa216-3197-11e8-82dd-e362eba9a59d.gif)
However I don't think we can change the title: **"The page at moz-extension:xxx-xxxx..."**

**Comment by vladikoff** [comment]
@ryanfeeley can we reuse footer notifications for this, and could you please provide text/copy for this?

**Comment by ryanfeeley** [comment]
@vladikoff @sebastienbarbier Is that dialog copy + the Reconnect to Sync footer style appropriate?
